From ec6f73ee3b3135a923c862f7e4bb431c9ca08cd8 Mon Sep 17 00:00:00 2001
From: Jaedon Shin <jaedon.shin@gmail.com>
Date: Thu, 4 Oct 2018 19:30:17 +0900
Subject: [PATCH] qcacld-2.0: Fix for 4.19

4f49dec9075aa0277b8c9c657ec31e6361f88724 net: allow ndo_select_queue to pass netdev

Signed-off-by: Jaedon Shin <jaedon.shin@gmail.com>
---
 CORE/HDD/inc/wlan_hdd_wmm.h  | 4 +++-
 CORE/HDD/src/wlan_hdd_main.c | 4 +++-
 CORE/HDD/src/wlan_hdd_wmm.c  | 4 +++-
 3 files changed, 9 insertions(+), 3 deletions(-)

diff --git a/CORE/HDD/inc/wlan_hdd_wmm.h b/CORE/HDD/inc/wlan_hdd_wmm.h
index 1634157cf5a0..9982614fa2c6 100644
--- a/CORE/HDD/inc/wlan_hdd_wmm.h
+++ b/CORE/HDD/inc/wlan_hdd_wmm.h
@@ -269,7 +269,9 @@ v_U16_t hdd_wmm_select_queue(struct net_device * dev, struct sk_buff *skb);
   ===========================================================================*/
 
 v_U16_t hdd_hostapd_select_queue(struct net_device * dev, struct sk_buff *skb
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(3,13,0))
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,19,0))
+                                 , struct net_device *sb_dev
+#elif (LINUX_VERSION_CODE >= KERNEL_VERSION(3,13,0))
                                  , void *accel_priv
 #endif
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(3,14,0))
diff --git a/CORE/HDD/src/wlan_hdd_main.c b/CORE/HDD/src/wlan_hdd_main.c
index 7af16df1bd67..93f8327223ca 100644
--- a/CORE/HDD/src/wlan_hdd_main.c
+++ b/CORE/HDD/src/wlan_hdd_main.c
@@ -10827,7 +10827,9 @@ static void hdd_set_multicast_list(struct net_device *dev)
   --------------------------------------------------------------------------*/
 static v_U16_t hdd_select_queue(struct net_device *dev,
                          struct sk_buff *skb
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(3,13,0))
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,19,0))
+                         , struct net_device *sb_dev
+#elif (LINUX_VERSION_CODE >= KERNEL_VERSION(3,13,0))
                          , void *accel_priv
 #endif
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(3,14,0))
diff --git a/CORE/HDD/src/wlan_hdd_wmm.c b/CORE/HDD/src/wlan_hdd_wmm.c
index f3f32b7d11ca..e7d940895400 100644
--- a/CORE/HDD/src/wlan_hdd_wmm.c
+++ b/CORE/HDD/src/wlan_hdd_wmm.c
@@ -1838,7 +1838,9 @@ hdd_wmm_classify_pkt(hdd_adapter_t* pAdapter, struct sk_buff *skb,
   @return         : Qdisc queue index
   ===========================================================================*/
 v_U16_t hdd_hostapd_select_queue(struct net_device * dev, struct sk_buff *skb
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(3,13,0))
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,19,0))
+                                 , struct net_device *sb_dev
+#elif (LINUX_VERSION_CODE >= KERNEL_VERSION(3,13,0))
                                  , void *accel_priv
 #endif
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(3,14,0))
-- 
2.19.1

